examples/cxx/publish_subscribe_dynamic_data/src/publisher.cpp

```diff
@@ -32,25 +32,25 @@ auto main() -> int {
                        .open_or_create()
                        .expect("successful service creation/opening");
 
-    uint64_t worst_case_memory_size = 1024; // NOLINT
-    auto publisher = service.publisher_builder()
-                         .max_slice_len(worst_case_memory_size)
-                         .create()
-                         .expect("successful publisher creation");
+    // Since the payload type is uint8_t, this number is the same as the number of bytes in the payload.
+    // For other types, number of bytes used by the payload will be max_slice_len * sizeof(Payload::ValueType)
+    const uint64_t maximum_elements = 1024; // NOLINT
+    auto publisher =
+        service.publisher_builder().max_slice_len(maximum_elements).create().expect("successful publisher creation");
 
-    auto counter = 1;
+    auto counter = 0;
 
     while (node.wait(CYCLE_TIME).has_value()) {
-        counter += 1;
-
-        auto required_memory_size = (8 + counter) % 16; // NOLINT
+        const auto required_memory_size = (counter % 16) + 1; // NOLINT
```

Comment:
The C++ and Rust version of the examples are identical in their functionality and structure, therefore could you please write here: `const auto required_memory_size = (8 + counter % 16)` just because it is handled like this in the rust example: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/examples/rust/publish_subscribe_dynamic_data/publisher.rs#L37

The idea is that soon you can start a C++ publisher and a Rust subscriber and the example works like it would have been pure C++ or pure Rust

Reply:
@elfenpiff I changed it because I thought was odd to have a payload size of 0 bytes. I can either revert it or change the rust example.

Reply:
Done